src/compiler/factory.ts

```diff
@@ -2299,6 +2299,28 @@ namespace ts {
         return node;
     }
 
+    export function createJsxText(text: string, containsOnlyWhiteSpaces: boolean) {
+        const node = <JsxText>createNode(SyntaxKind.JsxText);
+        node.text = text;
+        node.containsOnlyWhiteSpaces = containsOnlyWhiteSpaces;

```

Comment:
I doubt if we want to scan in the factory. 

ping @weswigham

Reply:
Yeah, I think we'd want the caller to make the determination, ideally. You need to know what the preceeding node is and where it ends to give the right answer, iirc - the text alone isn't actually what that flag is for. For example, the whitespace in `<div> </div>` isn't marked whitespace only because it's position makes it significant. TBH, I think the field name's a misnomer.